# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/front/wgsl/parse/directive/enable_extension.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/naga/src/front/wgsl/parse/directive/enable_extension.rs
--- a/third_party/rust/naga/src/front/wgsl/parse/directive/enable_extension.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/naga/src/front/wgsl/parse/directive/enable_extension.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -71,6 +71,7 @@
     const CLIP_DISTANCES: &'static str = "clip_distances";
     const DUAL_SOURCE_BLENDING: &'static str = "dual_source_blending";
     const SUBGROUPS: &'static str = "subgroups";
+    const PRIMITIVE_INDEX: &'static str = "primitive_index";
 
     /// Convert from a sentinel word in WGSL into its associated [`EnableExtension`], if possible.
     pub(crate) fn from_ident(word: &str, span: Span) -> Result<'_, Self> {
@@ -81,6 +82,9 @@
                 Self::Implemented(ImplementedEnableExtension::DualSourceBlending)
             }
             Self::SUBGROUPS => Self::Unimplemented(UnimplementedEnableExtension::Subgroups),
+            Self::PRIMITIVE_INDEX => {
+                Self::Unimplemented(UnimplementedEnableExtension::PrimitiveIndex)
+            }
             _ => return Err(Box::new(Error::UnknownEnableExtension(span, word))),
         })
     }
@@ -95,6 +99,7 @@
             },
             Self::Unimplemented(kind) => match kind {
                 UnimplementedEnableExtension::Subgroups => Self::SUBGROUPS,
+                UnimplementedEnableExtension::PrimitiveIndex => Self::PRIMITIVE_INDEX,
             },
         }
     }
@@ -132,12 +137,19 @@
     ///
     /// [`enable subgroups;`]: https://www.w3.org/TR/WGSL/#extension-subgroups
     Subgroups,
+    /// Enables the `@builtin(primitive_index)` attribute in WGSL.
+    ///
+    /// In the WGSL standard, this corresponds to [`enable primitive-index;`].
+    ///
+    /// [`enable primitive-index;`]: https://www.w3.org/TR/WGSL/#extension-primitive_index
+    PrimitiveIndex,
 }
 
 impl UnimplementedEnableExtension {
     pub(crate) const fn tracking_issue_num(self) -> u16 {
         match self {
             Self::Subgroups => 5555,
+            Self::PrimitiveIndex => 8236,
         }
     }
 }